### DISCUSSION TITLE
Theia custom theme

### DISCUSSION THREAD

**Discussion by quirinpa** [eclipse-theia/theia#10110] [Q&A] [answered]
Hello there. I've followed the tutorial at https://theia-ide.org/docs/composing_applications in order to get a Theia instance running with a custom theme in it. For this, I added the tomorrow night blue theme to my package.json's "theiaPlugins" section and I ran "yarn download:plugins" to download it. Then I copied its directory and made some changes. I then ran "yarn start". The plugin seems to be successfully loaded during boot. Then I visited "localhost:3000" and to my surprise all of the themes seem to be installed (including ones not included in my package.json), but I see no trace of my custom theme. What am I missing? Can you please help? Thank you.

**Comment by vince-fugnitto** [comment, marked as answer]
@quirinpa thank you for the discussion, I've never tried such a workflow before, I don't believe you can directly modify a packaged `.vsix` plugin like you've done.

You'll likely want to do the following which should work:
- follow the [create a theme](https://code.visualstudio.com/api/extension-guides/color-theme#create-a-new-color-theme) documentation
- you can use _tomorrow night blue_ as an example if you wish
- create your modifications
- package the plugin
- consume the plugin in theia

If you want to modify the theme and see the changes, you might also want to try the `hosted plugin` mode.

**Reply by quirinpa** [reply]
Thanks, we will try that procedure shortly. I'll get back to you. Cheers!

**Reply by quirinpa** [reply]
Still, isn't it strange that I got a lot of themes showing up when the only theme in my "theiaPlugins" section is tomorrow night blue?

**Reply by vince-fugnitto** [reply]
> Still, isn't it strange that I got a lot of themes showing up when the only theme in my "theiaPlugins" section is tomorrow night blue?

Which themes do you see?

**Reply by quirinpa** [reply]
> Which themes do you see?

Apparently it depends if I use a private chrome window or not. If I do use a private window, only Theia Light, Dark and High Contrast, and Tomorrow Night Blue appear. If I don't, all of the themes in the Theia repository are present: Theia's, Visual Studio's, Abyss, Dark+, Kimbie Dark, Monokai, Monokai Dimmed, Red, Solarized Dark, Tomorrow Night Blue and High Contrast.  I've ran Theia before using the main repository, maybe it is because of some leftover localstorage?

**Reply by vince-fugnitto** [reply]
Possibly, or if you're sharing `.theia` with multiple different application you might encounter issues as well.

**Reply by quirinpa** [reply]
Even using this procedure (directly adding a plugin to the plugins folder), I've managed to make it work. It turned out I was using an invalid base for the theme - "vs-light" instead of "vs". By 24th of September we will follow the correct procedure, thou, and we will get back to you, regardless. Thanks for the help.

**Reply by quirinpa** [reply]
Turns out our objective was a little different after all, because we were aiming to add the theme to a che-theia instance. We were able to do so without publishing the vsix file the usual way: We made a github release and linked to that in meta.yaml, linked to meta.yaml in che's yaml, and it worked. This is a bit off from the scope of the original question, so I'm just posting to tell you the conclusion of the situation. Thanks for your help!
